hazelcast/src/main/java/com/hazelcast/spi/discovery/multicast/MulticastDiscoveryStrategy.java

```diff
@@ -103,7 +100,7 @@ private void initializeMulticastSocket() {
             }
         } catch (Exception e) {
             logger.finest(e.getMessage());
-            rethrow(e);
+            throw rethrow(e);
```

Comment:
As far as I see `rethrow` also `throw`s, is it necessary to `throw` here?

Reply:
Not necessary, but IDE was showing some warnings. I did this to remove that warning

Reply:
OK. Maybe some code paths in `rethrow` doesn't throw the exception. If that's the case, it would be better to fix `rethrow` in another ticket.

Reply:
I think we can fit this little one into this PR